doc/developer/guide-changes.md

```diff
@@ -0,0 +1,790 @@
+be able to predict what kind of feedback you'll receive and preemptively address
+it, meaning you can embark on larger projects while seeking less up-front
+feedback. But you must still obey the [PR size limits](#pr-size-limits) for the
+sake of your reviewer!
+
+### Polish requirements
+
+PRs that are submitted for review should meet your own standards for
+"mergeable." Before you click the "Create PR" button, or immediately afterwards,
+do a self-review of your PR to make sure it is up to snuff.
+
+The self-review doesn't need to take more than a few minutes. The goal is to
+save your reviewer some time by filtering out glaring errors. Did you leave
+any comments like `// XXX: fix this before merging`? Did you leave any
+commented-out code or debugging `println!`s strewn about? Did you add any dead
+code? Did you add any blank lines to unrelated files? Did you remember to add
+all the tests you intended to? Have you made the PR as simple and small as it
+can be?
+
+The self-review is _not_ meant to add a large emotional burden before submitting
+a PR. You should not feel like PRs must be perfect before you can submit them!
+If your reviewers catch obvious typos or todos, that is not a failure—that's
+exactly why we have code review. The goal is only to reduce the probability that
+your reviewer notices glaring errors, not to drive the probability to zero.
+
+If you are intentionally looking for a review on a PR that is not mergeable—e.g.,
+because you are looking for [early feedback](#early-feedback) on high-level
+design questions—mark your PR as a [draft][draft-pr] so that your reviewers
+know to read past the unpolished parts.
+
+If your PR seems to meet the "mergeable" bar, but there are areas of the change
+that you are particularly unsure about, drop your reviewers a note on the PR
+indicating the danger zones so they can pay closer attention.
+
+### Choosing reviewers
+
+At the moment, choosing reviewers is a bit of a dark art. The best reviewer for
+a PR is the person besides you who is most familiar with the code that is
+changed. If you are not sure who that is, ask in #engineering.
+
+If you are more experienced, one additional aspect of choosing a reviewer to
+keep in mind is [mentoring](#mentoring). Asking a less experienced engineer to
+review your code is a great way for them to learn—often they'll pick up on
+tricks and habits that you've developed subconciously and might not think to
+mention in the reverse situation, when you're reviewing their code. This
+strategy works best when you are fairly confident in the change. If you're not
+confident in the change, consider assigning _both_ an experienced an
+inexperienced reviewer.
+
+Coming soon: a breakdown of which tech lead is responsible for each area of the
+codebase. The tech lead needn't review every PR in their area of responsibility,
+but can help direct code review requests to the appropriate engineer.
+
+### Merging
```

Comment:
Could we add a subsection on how to actually merge? (More specifically, how to rebase interactively versus the questionable Github "squash and merge" button?)

Reply:
There's a bit on using the "squash" strategy in the "Commit strategy" section, but not sure whether that really reflects your lived experience. Would you be up for making the adjustments you see fit once this merges?

Reply:
Yes!